tests/test_dojo.py

```diff
@@ -0,0 +1,289 @@
+		self.assertTrue(os.path.isfile("test_unallocated.txt"))
+		with open("test_unallocated.txt", "r") as outputfile:
+			lines = [line.rstrip() for line in outputfile]
+			self.assertTrue("PERSON NAME\t\tROLE" in lines)
+			self.assertTrue("George Wanjala\t\tFELLOW" in lines)
+		os.remove("test_unallocated.txt")
+
+class TestRellocatePerson(unittest.TestCase):
+	def setUp(self):
+		self.dojo = Dojo()
+
+	def test_rejects_inavalid_id(self):
+		rellocate_a = self.dojo.rellocate_person("a", "Nairobi")
+		self.assertEqual(rellocate_a, "\nInvalid Person ID\n")
+
+	def test_rejects_non_existent_id(self):
+		self.dojo.create_room("Nairobi", "Office")
+		self.dojo.add_person("fellow", "bob", "james")
+		self.dojo.create_room("Red", "Office")
+		rellocate_bob = self.dojo.rellocate_person(3, "Red")
+		self.assertEqual(rellocate_bob, "\nInvalid Person ID\n")
+	
+	def test_rejects_non_existent_room(self):
+		self.dojo.create_room("Nairobi", "Office")
+		self.dojo.add_person("fellow", "bob", "james")
+		rellocate_bob = self.dojo.rellocate_person(1, "Purple")
+		self.assertEqual(rellocate_bob, "\nNo such room exists\n")
+
+	def test_rellocates_a_person_succesfully_office(self):
+		self.dojo.create_room("Nairobi", "Office")
+		self.dojo.add_person("fellow", "bob", "james")
+		self.dojo.create_room("Kisii", "Office")
+		self.dojo.rellocate_person(1, "Kisii")
+		self.assertTrue(self.dojo.rooms[1].occupants)
+		self.assertFalse(self.dojo.rooms[0].occupants)
+
+	def test_rellocates_a_person_succesfully_livingspace(self):
+		self.dojo.create_room("Nairobi", "Office")
+		self.dojo.create_room("Kisii", "livingspace")
+		self.dojo.add_person("fellow", "bob", "james", "Y")
+		self.dojo.create_room("Kenya", "livingspace")
+		self.dojo.rellocate_person(1, "Kenya")
+		self.assertTrue(self.dojo.rooms[0].occupants)
+		self.assertFalse(self.dojo.rooms[1].occupants)
+		self.assertTrue(self.dojo.rooms[2].occupants)
+
+	def test_reject_reallocation_of_staff_to_livingspace(self):
+		self.dojo.create_room("Nairobi", "Office")
+		self.dojo.add_person("staff", "bob", "james")
+		self.dojo.create_room("Kisii", "Livingspace")
+		self.dojo.rellocate_person(1, "Kisii")
+		self.assertFalse(self.dojo.rooms[1].occupants)
+		self.assertTrue(self.dojo.rooms[0].occupants)
+
+	def test_rejects_reallocation_to_full_room(self):
+		self.dojo.create_room("Nairobi", "Office")
+		self.dojo.add_person("staff", "baba", "james")
+		self.dojo.add_person("staff", "bebe", "james")
+		self.dojo.add_person("staff", "bibi", "james")
+		self.dojo.add_person("staff", "bobo", "james")
+		self.dojo.add_person("staff", "bubu", "james")
+		self.dojo.add_person("staff", "babu", "james")
+		self.dojo.create_room("Kisii", "Office")
+		self.dojo.add_person("staff", "babe", "james")
+		relocate_babe = self.dojo.rellocate_person(7, "Nairobi")
+		msg ="The requested room is full. Try another room"
+		self.assertEqual(relocate_babe, msg)
+		self.assertTrue(self.dojo.rooms[0].occupants)
```

Comment:
As a suggestion , here you could also check that the room "Kisii" still has an occupant to make sure the occupant was not reallocated